allauth/account/adapter.py

```diff
@@ -483,15 +484,11 @@ def set_password(self, user, password):
 
     def get_user_search_fields(self):
         user = get_user_model()()
-        return filter(
-            lambda a: a and hasattr(user, a),
-            [
-                app_settings.USER_MODEL_USERNAME_FIELD,
-                "first_name",
-                "last_name",
-                "email",
-            ],
-        )
+        return [
+            f.name
+            for f in user._meta.fields
+            if type(f) in [CharField, EmailField] and f.name != "password"
```

Comment:
**Issue**: In this PR we're not just fixing the case where `first_name` is not there, but also making all text fields searchable by default, which may come a a performance penalty. I would like to stick to fixing just the former and not introduce additional fields to search.